libcaf_core/caf/actor_profiler.hpp

```diff
@@ -0,0 +1,76 @@
+/******************************************************************************
+ *                       ____    _    _____                                   *
+ *                      / ___|  / \  |  ___|    C++                           *
+ *                     | |     / _ \ | |_       Actor                         *
+ *                     | |___ / ___ \|  _|      Framework                     *
+ *                      \____/_/   \_|_|                                      *
+ *                                                                            *
+ * Copyright 2011-2019 Dominik Charousset                                     *
+ *                                                                            *
+ * Distributed under the terms and conditions of the BSD 3-Clause License or  *
+ * (at your option) under the terms and conditions of the Boost Software      *
+ * License 1.0. See accompanying files LICENSE and LICENSE_ALTERNATIVE.       *
+ *                                                                            *
+ * If you did not receive a copy of the license files, see                    *
+ * http://opensource.org/licenses/BSD-3-Clause and                            *
+ * http://www.boost.org/LICENSE_1_0.txt.                                      *
+ ******************************************************************************/
+
+#pragma once
+
+#include "caf/detail/build_config.hpp"
+#include "caf/fwd.hpp"
+
+namespace caf {
+
+/// A profiler which provides a set of callbacks for several actor operations in
+/// order to collect fine-grained profiling state about the system.
+/// @experimental
+class actor_profiler {
+public:
+  virtual ~actor_profiler();
+
+  /// Called whenever the actor system spawns a new actor. The system calls this
+  /// member function after the constructor of `self` has completed but before
+  /// constructing the behavior .
+  /// @param self The new actor.
+  /// @param parent Points to the parent actor unless `self` is a top-level
+  ///               actor (in this case, `parent` has the value `nullptr`).
+  /// @thread-safe
+  virtual void add_actor(const local_actor& self, const local_actor* parent)
+    = 0;
+
+  /// Called whenever the actor system spawns a new actor.
```

Comment:
"spawns a new actor"? copy/paste error from the add_actor description? :)

Reply:
Oops, definitely a copy & paste error. Thanks! 🙂